ev3dev/core.py

```diff
@@ -2361,6 +2367,38 @@ def raw(self):
 
         return self.value(0), self.value(1), self.value(2)
 
+    def calibrate_white(self):
+        """
+        The RGB raw values are on a scale of 0-1020 but you never see a value
+        anywhere close to 1020.  This function is designed to be called when
+        the sensor is placed over a white object in order to figure out what
+        are the maximum RGB values the robot can expect to see.  We will use
+        these maximum values to scale future raw values to a 0-255 range in
+        rgb().
+
+        If you never call this function red_max, green_max, and blue_max will
+        use a default value of 300.  This default was selected by measuring
+        the RGB values of a white sheet of paper in a well lit room.
+
+        Note that there are several variables that influene the maximum RGB
+        values detected by the color sensor
+        - the distance of the color sensor to the white object
+        - the amount of light in the room
+        - shadows that the robot casts on the sensor
+        """
+        (self.red_max, self.green_max, self.blue_max) = self.raw
+
+    @property
+    def rgb(self):
+        """
+        Same as raw() but RGB values are scaled to 0-255
+        """
+        (red, green, blue) = self.raw
+
+        return (min(int((red * 255) / self.red_max), 255),
+                min(int((green * 255) / self.green_max), 255),
+                min(int((blue * 255) / self.blue_max), 255))
```

Comment:
Scaling to red_max, green_max, blue_max instead of 1020 makes a HUGE difference in the accuracy of the RGB values returned.  I scanned my rubiks cube and displayed the colors returned from rgb() in a web page and they looked pretty good.  Before they all looked super super dark.